exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -209,9 +211,10 @@ public ParquetGroupScan( //
     this.entries = Lists.newArrayList();
     if (fileSelection.getMetaContext() != null &&
         (fileSelection.getMetaContext().getPruneStatus() == PruneStatus.NOT_STARTED ||
-          fileSelection.getMetaContext().getPruneStatus() == PruneStatus.NOT_PRUNED)) {
-      // if pruning was not applicable or was attempted and nothing was pruned, initialize the
-      // entries with just the selection root instead of the fully expanded list to reduce overhead.
+          fileSelection.getMetaContext().getPruneStatus() == PruneStatus.NOT_PRUNED ||
+            fileSelection.getMetaContext().isMetadataFilesMissingOrCorrupted)) {
```

Comment:
There are 4 checks in `if` statement, may be we can factor out them in separate method?